packages/http/src/client.ts

```diff
@@ -0,0 +1,142 @@
+import { IPrismOutput } from '@stoplight/prism-core';
+import { IHttpOperation } from '@stoplight/types';
+// @ts-ignore
+import logger from 'abstract-logging';
+import { getOrElse } from 'fp-ts/lib/Option';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { getStatusText } from 'http-status-codes';
+import { defaults, partial } from 'lodash';
+import { parse as parseQueryString } from 'querystring';
+import { parse as parseUrl } from 'url';
+import { createInstance } from '.';
+import { forwarder } from './forwarder';
+import getHttpOperations, { getHttpOperationsFromResource } from './getHttpOperations';
+import { mocker } from './mocker';
+import { router } from './router';
+import { IHttpConfig, IHttpRequest, IHttpResponse, IHttpUrl } from './types';
+import { validator } from './validator';
+
+function createClientFrom(
+  getResource: (v: string) => Promise<IHttpOperation[]>,
+  spec: string,
+  defaultConfig: IHttpConfig,
+): Promise<PrismHttp> {
+  return getResource(spec).then(resources => createClientFromOperations(resources, defaultConfig));
+}
+
+const createClientFromResource = partial(createClientFrom, getHttpOperationsFromResource);
+const createClientFromString = partial(createClientFrom, getHttpOperations);
+
+function createClientFromOperations(resources: IHttpOperation[], defaultConfig: IHttpConfig) {
+  const obj = createInstance(defaultConfig, {
+    logger,
+    router,
+    forwarder,
+    validator,
+    mocker,
+  });
+
+  const request: RequestFunction = async (url, input, config) => {
+    const parsedUrl = parseUrl(url);
+
+    if (!parsedUrl.pathname) throw new Error('path name must alwasy be specified');
+
+    const httpUrl: IHttpUrl = {
+      baseUrl: parsedUrl.host ? `${parsedUrl.protocol}//${parsedUrl.host}` : undefined,
+      path: parsedUrl.pathname,
+      query: parseQueryString(parsedUrl.query || ''),
+    };
+
+    const data = await obj.request(
+      {
+        ...input,
+        url: httpUrl,
+      },
+      resources,
+      config,
+    );
+
+    const output: PrismOutput = {
+      status: data.output.statusCode,
+      statusText: pipe(
+        data.output.statusText,
```

Comment:
This is not a thing anymore. You can ignore it.

To explain why this was needed — Axios will return the statusText if provided; but the Prism mocker won't. For this reason, we needed a default in case the response was coming from the mocker. Does this clarify?